controller/lighting_unit.py

```diff
@@ -0,0 +1,140 @@
+from neopixel import NeoPixel
+from machine import Pin
+
+
+class LightingUnit:
+    """A class representing an LED lighting unit. It implements some basic
+    functionality necessary for controlling a hardware LED lighting unit,
+    such as those made up of addressable pixels (like the WS2812B) connected
+    in serial
+
+    :param outputPin: the pin ID for the controller's output to the lighting
+        unit
+    :type outputPin: int
+    :param pixelCount: the number of pixels in the lighting unit
+    :type pixelCount: int
+    """
+    def __init__(self, outputPin, pixelCount):
```

Comment:
Python allows type hints: https://peps.python.org/pep-0484/

I would strongly recommend putting them in your code.  It saves on the documentation.  Example:

`def __init__(self, outputPin: int, pixelCount: int`

Type hints can be in the parameter list or in the variable itself.

self._foo: int = 0